collector/src/credential.rs

```diff
@@ -0,0 +1,105 @@
+use crate::Error;
+use hpke_dispatch::{Aead, Kdf, Kem};
+use janus_core::{
+    auth_tokens::{AuthenticationToken, BearerToken},
+    hpke::{self, HpkeKeypair, HpkePrivateKey},
+};
+use janus_messages::{HpkeConfig, HpkeConfigId, HpkePublicKey};
+use serde::{Deserialize, Serialize};
+
+/// Serializable representation of a private collector credential, which contains all secrets a
+/// collector uses to interact with an aggregator. It contains the [`BearerToken`] for
+/// authorization to an aggregator, and the private HPKE configuration for decrypting aggregate
+/// shares.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PrivateCollectorCredential {
+    id: HpkeConfigId,
+    kem: Kem,
+    kdf: Kdf,
+    aead: Aead,
+    public_key: HpkePublicKey,
+    private_key: HpkePrivateKey,
+    // TODO(inahga): Turn this non-optional for the next breaking change cycle.
+    token: Option<BearerToken>,
```

Comment:
Unsure if this should be Option or not.

This comes from an iteration of `divviup` where it output credentials without an associated token. I don't think that made it to a release and therefore I'm not sure if any such credentials are out in the wild. In the strictest of terms it should be Option for backwards-compatibility, but unsure if being too strict here is counterproductive.

Reply:
IMO: even if we didn't generate such credentials, I suppose we can't be sure that no one did. I think we should make the conservative choice to keep this optional for now, and write up an issue tracking our desire to make this non-optional with the next time we make breaking changes.